feat: Extend CAST to ARRAY, MAP and STRUCT (#6587)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/scalar-functions.md

   GROUP BY ID;
```

### CAST

> Since: -

Reviewer: Since 0.15.0?

Author: CAST has always been there...

```sql
CAST(COL0 AS BIGINT)
```

Converts one type to another. The following casts are supported:

| from | to | notes |
|------|----|-------|
| ANY  | STRING | Converts the type to its string representation. |

> | STRING | BOOLEAN | Any string that exactly matches `true`, case-insensitive, will be converted to `true`. Any other value will be converted to `false`. |

Reviewer: ```suggestion
| STRING | BOOLEAN | Any string that exactly matches `true`, case-insensitive, is converted to `true`. Any other value is converted to `false`. |
```

> | STRING | INT, BIGINT, DECIMAL, DOUBLE | Converts string representation of numbers to number types. Conversion will fail if text does not contain a number or the number does not fit in the indicated type. |

Reviewer: ```suggestion
| STRING | INT, BIGINT, DECIMAL, DOUBLE | Converts string representation of numbers to number types. Conversion fails if text doesn't contain a number or the number doesn't fit in the indicated type. |
```

Reviewer: Does "fail" mean throw an exception?

Author: Sure, but 'throws an exception' doesn't necessarily mean anything to the user of ksqlDB.  This isn't a library where the dev using it needs to know to expect an exception.

| INT, BIGINT, DECIMAL, DOUBLE | INT, BIGINT, DECIMAL, DOUBLE | Convert between numeric types. Conversion can result in rounding |
| ARRAY | ARRAY | Convert between arrays of different element types |   
| MAP | MAP | Convert between maps of different key and value types |   

> | STRUCT | STRUCT | Convert between structs of different field types. Only fields that existing in the target STRUCT type will be copied across. Any fields in the target type that do not exist in the source will be set to `NULL` |

Reviewer: ```suggestion
| STRUCT | STRUCT | Convert between structs of different field types. Only fields that exist in the target STRUCT type are copied across. Any fields in the target type that don't exist in the source are set to `NULL` |
```
